src/test/java/com/teragrep/pth_06/planner/CategoryTableImplTest.java

```diff
@@ -173,11 +169,24 @@ public void testCreation() {
                 .getTables()
                 .get(0);
 
-        TokenizedValue val = new TokenizedValue("test");
-        CategoryTable created = new CategoryTableImpl(ctx, table, 0L, val);
-        created.create();
-        Assertions
-                .assertEquals(created.bloommatchCondition(), new CategoryTableImpl(ctx, table, 0L, val).bloommatchCondition());
+        CategoryTable categoryTable = new CategoryTableImpl(ctx, table, 0L, "test");
+        Assertions.assertDoesNotThrow(categoryTable::create);
+    }
+
+    @Test
+    public void testFilterInsertion() {
+        fillTargetTable();
+        DSLContext ctx = DSL.using(conn);
+        Table<?> table = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+
+        CategoryTable categoryTable = new CategoryTableImpl(ctx, table, 0L, "192.168.1.1");
+        Assertions.assertDoesNotThrow(categoryTable::create);
```

Comment:
could this use assertion that the filters are inserted?

Reply:
Yes good spot I updated test to check that only pattern matching tokens are inserted. Changed TokenizedValue to find minor tokens since for example search term `ip=192.168.1.1` did not get token `192.168.1.1`.